tools/tesseratos/include/tesseratos/toolbox/plugin.hpp

```diff
@@ -0,0 +1,61 @@
+
+
+
+/// @dir
+/// @brief @ref toolbox-plugin plugin directory.
+
+/// @file
+/// @brief Plugin entry point.
+/// @ingroup toolbox-plugin
+
+#pragma once
+
+#include <cubos/engine/cubos.hpp>
+
+
+
+
+namespace tesseratos
+{
+    /// @defgroup tesseratos-toolbox-plugin Toolbox
+    /// @ingroup tesseratos
+    /// @brief Adds a resource used to keep track of whether each tool is open or not.
+    ///
+    ///
+    /// ## Resources
+    /// - @ref Toolbox 
+
+    /// @brief Resource which manages other tools windows.
+    class Toolbox final
+    {
+        public:
+
+        std::map<std::string, bool> toolsMap;
```

Comment:
I think we don't care about order, and `std::unordered_map` is faster(?) (or maybe does not matter at all)